worldedit-forge/src/main/java/com/sk89q/worldedit/forge/ForgeEntity.java

```diff
@@ -84,19 +85,19 @@ public boolean setLocation(Location location) {
 
     @Override
     public Extent getExtent() {
-        net.minecraft.entity.Entity entity = entityRef.get();
+        net.minecraft.world.entity.Entity entity = entityRef.get();
         if (entity != null) {
-            return ForgeAdapter.adapt(entity.world);
+            return ForgeAdapter.adapt((ServerLevel) entity.level);
         } else {
             return NullWorld.getInstance();
         }
     }
 
     @Override
     public boolean remove() {
-        net.minecraft.entity.Entity entity = entityRef.get();
+        net.minecraft.world.entity.Entity entity = entityRef.get();
         if (entity != null) {
-            entity.remove();
+            entity.remove(net.minecraft.world.entity.Entity.RemovalReason.KILLED);
```

Comment:
Is this the right removal reason?

Reply:
It's what I used with Fabric. The other option is `DISCARDED`, but I think we would rather fire the kill event since `remove()` is a lot like `/kill`.

In the future we should probably add a removal reason of our own, to mark cut entities as "discarded"? Not sure about that but it sorta makes sense.